src/opencl/mod256-189.cu

```diff
@@ -0,0 +1,970 @@
+typedef ulong limb_t;
+
+typedef limb_t vec256[256/(8*sizeof(limb_t))];
+
+struct vec256_t {
+    union { vec256 l;
+            uint v[sizeof(vec256)/sizeof(uint)];
+    };
+    
+};
+
+//extern __shared__ uint32_t scratchpad[];    // dynamically allocated
+
+//asm volatile(".extern .shared .b32 scratchpad[];\t\n");
+
+struct vec256_shared_t {
```

Comment:
Again, some basic comments would be beneficial for an outsider to understand, or even for us to remember what is the purpose here. I assume this struct is for shared memory, v0 is storing the vector itself, and v_off is the offset, right? But if I'm wrong, it will be very hard for me to understand the rest of the code. I think it will be nice to have this explanatory comments for every struct defined